gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ApiExtensionController.java

```diff
@@ -0,0 +1,81 @@
+package com.netflix.spinnaker.gate.controllers;
+
+import static org.springframework.web.bind.annotation.RequestMethod.*;
+
+import com.netflix.spinnaker.gate.api.extension.ApiExtension;
+import com.netflix.spinnaker.gate.api.extension.HttpRequest;
+import com.netflix.spinnaker.gate.api.extension.HttpResponse;
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import okhttp3.internal.http.HttpMethod;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * A top-level {@code RestController} that exposes all api extensions under a common
+ * `/extensions/{{ApiExtension.id}}` path.
+ */
+@Alpha
+@RestController
+@RequestMapping("/extensions")
+public class ApiExtensionController {
+  private List<ApiExtension> apiExtensions;
+
+  @Autowired
+  public ApiExtensionController(List<ApiExtension> apiExtensions) {
+    this.apiExtensions = apiExtensions;
+  }
+
+  @RequestMapping(value = {"/{extension}/**"})
```

Comment:
Not sure how we feel about restricting this api extensions to a common path prefix or allowing them anywhere?

The dynamic nature of these endpoints aren't self-documenting, so that's a draw-back (similar with the `ProxyController`).

Reply:
I prefer restricting to a common path prefix for now - I can't think of a great reason not to do that.